q2_demux/test/test_demux.py

```diff
@@ -1,216 +1,196 @@
         with self.assertRaises(ValueError):
             list(bsi)
 
 
 class EmpTests(unittest.TestCase):
 
     def setUp(self):
-        barcodes = [
-            skbio.DNA('AAAA', metadata={'id': 's1/2', 'description': 'abc/2'},
-                      positional_metadata={'quality': [22, 25, 22, 18]}),
-            skbio.DNA('AAAA', metadata={'id': 's2/2', 'description': 'abc/2'},
-                      positional_metadata={'quality': [25, 25, 25, 25]}),
-            skbio.DNA('AACC', metadata={'id': 's3/2', 'description': 'abc/2'},
-                      positional_metadata={'quality': [22, 25, 22, 18]}),
-            skbio.DNA('AACC', metadata={'id': 's4/2', 'description': 'abc/2'},
-                      positional_metadata={'quality': [22, 25, 22, 18]}),
-        ]
-        # setting the dtype on these quality scores as they end up being
-        # compared to the output, which gets written as uint8
-        self.sequences = [
-            skbio.DNA('GGG', metadata={'id': 's1/1', 'description': 'abc/1'},
-                      positional_metadata={'quality':
-                      np.array([22, 25, 22], dtype='uint8')}),
-            skbio.DNA('CCC', metadata={'id': 's2/1', 'description': 'abc/1'},
-                      positional_metadata={'quality':
-                      np.array([29, 29, 29], dtype='uint8')}),
-            skbio.DNA('AAA', metadata={'id': 's3/1', 'description': 'abc/1'},
-                      positional_metadata={'quality':
-                      np.array([22, 20, 22], dtype='uint8')}),
-            skbio.DNA('TTT', metadata={'id': 's4/1', 'description': 'abc/1'},
-                      positional_metadata={'quality':
-                      np.array([22, 25, 22], dtype='uint8')}),
-        ]
-        self.bsi = BarcodeSequenceIterator(barcodes, self.sequences)
-
-        barcodes = pd.Series(['AAAA', 'AACC'], index=['sample1', 'sample2'])
-        self.barcodes = qiime.MetadataCategory(barcodes)
+        barcodes = [('@s1/2 abc/2', 'AAAA', '+', 'YYYY'),
+                    ('@s2/2 abc/2', 'AAAA', '+', 'PPPP'),
+                    ('@s3/2 abc/2', 'AACC', '+', 'PPPP'),
+                    ('@s4/2 abc/2', 'AACC', '+', 'PPPP')]
+
+        self.sequences = [('@s1/1 abc/1', 'GGG', '+', 'YYY'),
+                          ('@s2/1 abc/1', 'CCC', '+', 'PPP'),
+                          ('@s3/1 abc/1', 'AAA', '+', 'PPP'),
+                          ('@s4/1 abc/1', 'TTT', '+', 'PPP')]
+        self.bsi = BarcodeSequenceFastqIterator(barcodes, self.sequences)
+
+        barcode_map = pd.Series(['AAAA', 'AACC'], index=['sample1', 'sample2'])
+        self.barcode_map = qiime.MetadataCategory(barcode_map)
+
+    def _decode_qual_to_phred(self, qual_str):
+        # this function is adapted from scikit-bio
+        qual = np.fromstring(qual_str, dtype=np.uint8) - 33
+        return qual
+
+    def _compare_sequence_to_fields(self, sequence, fields):
+        header_line = ' '.join([sequence.metadata['id'],
```

Comment:
This is confusing, can you just check the record lines against their expected values?